htlcswitch/interceptable_switch.go

```diff
@@ -390,6 +408,23 @@ func (s *InterceptableSwitch) resolve(res *FwdResolution) error {
 
 // Resolve resolves an intercepted packet.
 func (s *InterceptableSwitch) Resolve(res *FwdResolution) error {
+	switch {
+	// Verify that for encrypted messages the size is equal to the fixed
+	// failure message size + hmac + two uint16 lengths. See BOLT #4.
+	case res.FailureMessageEncrypted && len(res.FailureMessage) !=
+		lnwire.FailureMessageLength+32+2+2:
+
+		return ErrInvalidFailureMsgLength
+
+	// For unencrypted messages, verify that they are parseable.
+	case !res.FailureMessageEncrypted:
+		r := bytes.NewReader(res.FailureMessage)
+		_, err := lnwire.DecodeFailureMessage(r, 0)
+		if err != nil {
+			return ErrUnparseableFailureMsg
+		}
```

Comment:
I want to add extra code here to re-sign a channel update with this node's key if it is part of the decoded failure message.

Reply:
Tacked on commit for this.